### PR TITLE
fixed Download Error on Some Devices

### LINKED ISSUE #6: Download error | apk 1.2
on click the option "DOWNLOAD AND INSTALL SERVER.ZIP" and "TOGGLE AUTHENTICATION"

screenshot_2017-03-13-20-20-33

screenshot_2017-03-13-20-26-07

### PR DESCRIPTION
**Pull request by mrbesen**
This Fixed #6  for me.

### CODE REVIEW

File: app/src/main/java/org/cuberite/android/MainActivity.java [hunk 1]
```diff
@@ -326,6 +326,8 @@ public void onClick(DialogInterface dialog, int id) {
             builder.create().show();
         } else if(preferences.getString("cuberiteLocation", "").startsWith(PRIVATE_DIR)){
             Log.d(Tags.MAIN_ACTIVITY, "Cuberite has permissions for external storage, but is still in the private dir");
+        } else if(preferences.getString("cuberiteLocation", null) == null) {
+            preferences.edit().putString("cuberiteLocation", "/sdcard/cuberite-server").apply();
```
Comment (madmaxoft) [inline app/src/main/java/org/cuberite/android/MainActivity.java]: I don't think a `checkPermission` function is a good place to be overriding any preferences.

Comment (mrbesen) [reply app/src/main/java/org/cuberite/android/MainActivity.java]: yeah, that is not a nice decision.
But the only place where i found, that preferences are set, is at this function: 
```onRequestPermissionsResult```
but this function is only indirectly called by ```checkPermission``` so the function is changing the preferences already.

Comment (mrbesen) [reply app/src/main/java/org/cuberite/android/MainActivity.java]: but maybe its useful to change the hardcoded path to something like this
```suggestion
preferences.edit().putString("cuberiteLocation", PUBLIC_DIR + "/cuberite-server").apply();
```

**Review by ItsShadowCone — CHANGES_REQUESTED** [review]
I agree with xoft here that `checkPermissions` is not the right place. Since this is roughly modeling a finite state machine I would accept an intermediate solution as a `verifySettings()` called on button presses inside checkState (which is poorly named our state machine loop basically) akin to how checkPermissions is called in doStart (validating a correct settings structure shortly before its actually used). This might want to be placed as a static in SettingsActivity (akin to initializeSettings).

Aside from that is the discussion what should happen with the Android app at all, I kind of owe cuberite a complete rewrite (rediscovering this code after all this time shows me how poorly it was implemented), also news to most of us here is that finally after I think a year of noncompliance Google has removed the app from the play store. Self-publishing / publishing via fdroid (with caveats) is possible nonetheless. But let's move to the forums for that discussion.

**Pull request closed without merging**